sections/main-addresses.liquid

```diff
@@ -27,92 +36,125 @@
     <a href="{{ routes.account_url }}">
       {{ 'customer.account.return' | t }}
     </a>
-
-    <div data-address>
+    {%- form 'customer_address', customer.new_address, aria-labelledBy: 'AddressNewHeading', data-address: '' -%}
       <button
         type="button"
-        aria-expanded="false"
+        aria-expanded="{% if form.errors %}true{% else %}false{% endif %}"
         aria-controls="AddAddress"
       >
         {{ 'customer.addresses.add_new' | t }}
       </button>
-      <div id="AddAddress">
+      <div id="AddAddress" class="address-form-fields">
         <h2 id="AddressNewHeading">{{ 'customer.addresses.add_new' | t }}</h2>
-        {%- form 'customer_address', customer.new_address, aria-labelledBy: 'AddressNewHeading' -%}
-          <div class="field">
-            <input type="text" id="AddressFirstNameNew" name="address[first_name]" value="{{ form.first_name }}" autocomplete="given-name" placeholder="{{ 'customer.addresses.first_name' | t }}">
-            <label for="AddressFirstNameNew">{{ 'customer.addresses.first_name' | t }}</label>
-          </div>
-          <div class="field">
-            <input type="text" id="AddressLastNameNew" name="address[last_name]" value="{{ form.last_name }}" autocomplete="family-name" placeholder="{{ 'customer.addresses.last_name' | t }}">
-            <label for="AddressLastNameNew">{{ 'customer.addresses.last_name' | t }}</label>
+        {%- if form.errors -%}
+          <div>
+            <h3 class="form__message" tabindex="-1" autofocus>
```

Comment:
> then the screen reader automatically starts reading out the page contents from the beginning

Do you have "Automatically speak the webpage" option enabled for VoiceOver? I think disabling the option would prevent that behaviour.

![Screen Shot 2022-11-14 at 12 49 25 PM](https://user-images.githubusercontent.com/3710548/201664458-110cb63d-3990-45b7-91e9-fa762d1fb27d.png)

Reply:
> Page updates like this should be announced to screen reader users via aria-live or a role attribute

This is true for parts of the page that update dynamically without a page refresh. Our form submission triggers a full page reload and according to this [thread](https://github.com/w3c/aria/issues/1360), it appears relying on aria live regions after page reload produces inconsistent behaviour in screenreaders. 

This is why we've typically opted to add `autofocus` for our form errors. This ensures that when the page is reloaded, focus is automatically moved to the form error heading. I believe the `tabindex="-1"` is also necessary, otherwise the browser will ignore the `autofocus` attribute.

I am noticing that after page reload, the browser moves focus to the form error heading, but then switches to another part of the page shortly after. I'm currently looking into what's causing this.

Reply:
Thanks for explaining. I didn't realise we're doing a full page refresh here, so that all makes sense! Even without automatically speaking the webpage, the focus didn't stay on the error message. But it sounds like you've also spotted the same issue 👍

Reply:
Update - https://github.com/Shopify/dawn/pull/2118/commits/cef5205cdd33c7c37521b7f1aea461a03ebc79cb. It appears despite having the `autofocus` attribute, the form error heading isn't always fully announced by VO on page refresh. VO seems to shift its focus to the page's `main` element instead. After some trial and error, I found that changing the heading element to `span` yields more consistent results in Chrome atleast. I also added `role="alert"` to prevent VO from identifying the `span` as a "group".

Reply:
That's interesting, because semantically it should probably be a heading 🤔 Is this what we're doing in other forms, too, using `span` instead of a heading?

Reply:
I think keeping the heading and moving the focus to `span` is a good compromise. There's value in allowing users to easily discover this section if they navigate by heading level.

Reply:
I reverted back to `h3`, but seems like moving the `autofocus` to `span` does not really fix the VO issue. 😢

I'm seeing [reports of inconsistent behaviour with VO and `autofocus` in general](https://bugs.webkit.org/show_bug.cgi?id=163719#c2), so this is likely not isolated to this scenario. I would probably open a new issue and keep investigating this separate from this PR.